tests/src/Simple/Generics/Generics.cs

```diff
@@ -567,4 +568,221 @@ public static void Run()
+            string IFoo<int>.IMethod1<T>(T t1, T t2) { return "MyStruct1.IFoo<int>.IMethod1<" + typeof(T) + ">(" + t1 + "," + t2 + ")"; }
+        }
+        struct MyStruct2 : IFoo<string>, IFoo<int>
+        {
+            string IFoo<string>.IMethod1<T>(T t1, T t2) { return "MyStruct2.IFoo<string>.IMethod1<" + typeof(T) + ">(" + t1 + "," + t2 + ")"; }
+            public string IMethod1<T>(T t1, T t2) { return "MyStruct2.IMethod1<" + typeof(T) + ">(" + t1 + "," + t2 + ")"; }
+        }
+        struct MyStruct3 : IFoo<string>, IFoo<int>
+        {
+            string IFoo<int>.IMethod1<T>(T t1, T t2) { return "MyStruct3.IFoo<int>.IMethod1<" + typeof(T) + ">(" + t1 + "," + t2 + ")"; }
+            public string IMethod1<T>(T t1, T t2) { return "MyStruct3.IMethod1<" + typeof(T) + ">(" + t1 + "," + t2 + ")"; }
+        }
+
+        static string s_GMethod1;
+        static string s_IFooString;
+        static string s_IFooObject;
+        static string s_IFooInt;
+
+        static int s_NumErrors = 0;
+
+        private static void TestWithStruct(IFoo<string> ifooStr, IFoo<object> ifooObj, IFoo<int> ifooInt)
+        {
+            var res = ifooStr.IMethod1<int>(1, 2);
+            WriteLineWithVerification(res, s_IFooString);
+
+            res = ifooObj.IMethod1<int>(3, 4);
+            WriteLineWithVerification(res, s_IFooObject);
+
+            res = ifooInt.IMethod1<int>(5, 6);
+            WriteLineWithVerification(res, s_IFooInt);
+        }
+
+        private static void TestWithClass(object o)
+        {
+            Base b = o as Base;
+            var res = b.GMethod1<int>(1, 2);
+            WriteLineWithVerification(res, s_GMethod1);
+
+            IFoo<string> ifoo1 = o as IFoo<string>;
+            res = ifoo1.IMethod1<int>(3, 4);
+            WriteLineWithVerification(res, s_IFooString);
+
+            IFoo<object> ifoo2 = o as IFoo<object>;
+            res = ifoo2.IMethod1<int>(5, 6);
+            WriteLineWithVerification(res, s_IFooObject);
+
+            IFoo<int> ifoo3 = o as IFoo<int>;
+            res = ifoo3.IMethod1<int>(7, 8);
+            WriteLineWithVerification(res, s_IFooInt);
+        }
+
+        private static void TestWithGenClass<T>(object o)
+        {
+            GenBase<T> b = o as GenBase<T>;
+            var res = b.GMethod1<int>(1, 2);
+            WriteLineWithVerification(res, s_GMethod1);
+
+            IFoo<string> ifoo1 = o as IFoo<string>;
+            res = ifoo1.IMethod1<int>(3, 4);
+            WriteLineWithVerification(res, s_IFooString);
+
+            IFoo<object> ifoo2 = o as IFoo<object>;
+            res = ifoo2.IMethod1<int>(5, 6);
+            WriteLineWithVerification(res, s_IFooObject);
+
+            IFoo<int> ifoo3 = o as IFoo<int>;
+            res = ifoo3.IMethod1<int>(7, 8);
+            WriteLineWithVerification(res, s_IFooInt);
+
+            res = b.VMethod1();
+            Console.WriteLine(res);
```

Comment:
I'll remove this non-generic virtual method (i added it a while back when i needed to debug the non-generic virtual path)